packages/flutter/lib/src/widgets/scroll_physics.dart

```diff
@@ -500,6 +552,14 @@ class ClampingScrollPhysics extends ScrollPhysics {
     return 0.0;
   }
 
+  @override
+  double adjustPositionForNewDimensions(ScrollMetrics position, {
+    @required bool isScrolling,
+    @required double velocity,
+  }) {
+    return position.pixels.clamp(position.minScrollExtent, position.maxScrollExtent) as double;
```

Comment:
when I was noodling on this I had assumed that if we do it even when velocity is zero, it will cause animations to flail because the animation will immediately try to snap it back... is that not the case?

Reply:
The system is complicated enough that I'm trying to move to being test driven and ripped out assumptions.  Those needed to be ripped out specifically to get dragging working.  I can add a test for shrinking the view while it's ballistic.

Reply:
I was just playing around with the stepper.  Looks like if you animate the resizing of you scrollview this will just poke the animation to make sure it stays in bounds.  Jumping might not be erroneous then if you abruptly change the size, if you want it smooth, use an animation.